dagmc_stats/dagmc_stats.py

```diff
@@ -471,12 +471,71 @@ def get_roughness(my_core, native_ranges):
 
     outputs
     -------
-    roughness : (numpy array) the roughness for all surfaces in the meshset
+    roughness : (dictionary) the roughness for all vertices in the meshset
     """
     tri_vert_data, all_verts = get_tri_vert_data(my_core,
                                                     native_ranges[types.MBTRI])
     gc_all = get_gaussian_curvature(my_core, all_verts, tri_vert_data)
-    roughness = np.zeros(len(all_verts))
-    for vert_idx, vert_i in enumerate(all_verts):
-        roughness[vert_idx] = get_lri(vert_i, gc_all, tri_vert_data, my_core)
+    roughness = {}
+    for vert_i in all_verts:
+        roughness[vert_i] = get_lri(vert_i, gc_all, tri_vert_data, my_core)
+    if tag:
+        #add triangle average roughness tag
+        lr_tag = {}
+        for tri in native_ranges[types.MBTRI]:
+            three_verts = list(my_core.get_adjacencies(tri, 0, op_type=1))
+            sum_lr = 0
+            for vert in three_verts:
+                sum_lr += roughness[vert]
+            lr_tag[tri] = sum_lr/3
+        add_tag(my_core, 'FACET_AVG_LR', lr_tag, types.MB_TYPE_DOUBLE)
     return roughness
+    
+    
+def add_tag(my_core, tag_name, tag_dic, tag_type):
```

Comment:
this function needs a test- we can discuss if you don't have an idea for how to test this.

Reply:
Also kudos on this function- This will be a good one for reusing in many contexts.